sql/analyzer/hoist_select_exists.go

```diff
@@ -160,6 +160,12 @@ func hoistExistSubqueries(ctx *sql.Context, scope *plan.Scope, a *Analyzer, filt
 			}
 		}
 
+		if sqa, ok := s.inner.(*plan.SubqueryAlias); ok {
+			if !sqa.CanCacheResults() {
```

Comment:
wondering if it's possible for inner to not be a SQA, but still have a lateral reference. something like `SELECT c_id, bill FROM c AS c2 WHERE EXISTS(SELECT c_id, coalesce(ship, bill) AS state FROM o WHERE c_id = c2.c_id AND state = bill);`

Reply:
Can't have that query exactly cause of MySQL's aliasing rules, but this should be equivalent
````
SELECT c_id, bill FROM c AS c2 WHERE EXISTS(SELECT c_id, ship FROM o WHERE c_id = c2.c_id AND coalesce(ship, bill) = bill);
```